pkg/flow/internal/testcomponents/singleton.go

```diff
@@ -0,0 +1,59 @@
+package testcomponents
+
+import (
+	"context"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/component"
+)
+
+func init() {
```

Comment:
Discussion point related to my above comment: IMO Singleton is fine for now, but I feel submodules will force us to define this more concretely or use different verbiage.